[WebGPU] Update WebGPU.h to latest revision - fa1c6ab4927ef1fa5731907b42b62ea93119866c (250856)

mwyrzykowski opened the pull request:
#### 419ae1b379f760ca7aad4d6928864007169f9be4
<pre>
[WebGPU] Update WebGPU.h to latest revision - fa1c6ab4927ef1fa5731907b42b62ea93119866c (250856)
<a href="https://bugs.webkit.org/show_bug.cgi?id=250856">https://bugs.webkit.org/show_bug.cgi?id=250856</a>
&lt;radar://104437950&gt;

Reviewed by Myles C. Maxfield.

Pull the latest version of WebGPU.h and update our
sources to use it.

Significant changes:
 * Shader hints moved into WebGPU.h, previously in WebGPUExt.h
 * MipmapFilter became its own type
 * GPUAdapter has requestAdapterInfo

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/WebGPU/GPUAdapter.cpp:
(WebCore::GPUAdapter::requestAdapterInfo):
* Source/WebCore/Modules/WebGPU/GPUAdapter.h:
* Source/WebCore/Modules/WebGPU/GPUAdapter.idl:
* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUFeatureName.h:
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUFeatureName.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerDescriptor.h:
* Source/WebCore/Modules/WebGPU/GPUSamplerDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureFormat.h:
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUTextureFormat.idl:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUAdapterImpl.cpp:
(PAL::WebGPU::supportedFeatures):
(PAL::WebGPU::AdapterImpl::requestDevice):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUComputePassEncoderImpl.cpp:
(PAL::WebGPU::ComputePassEncoderImpl::dispatch):
(PAL::WebGPU::ComputePassEncoderImpl::dispatchIndirect):
(PAL::WebGPU::ComputePassEncoderImpl::end):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.cpp:
(PAL::WebGPU::ConvertToBackingContext::convertToBacking):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.cpp:
(PAL::WebGPU::createBackingDescriptor):
(PAL::WebGPU::DeviceImpl::createShaderModule):
(PAL::WebGPU::DeviceImpl::popErrorScope):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUQueueImpl.cpp:
(PAL::WebGPU::QueueImpl::onSubmittedWorkDone):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUQueueImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPURenderPassEncoderImpl.cpp:
(PAL::WebGPU::RenderPassEncoderImpl::end):
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUFeatureName.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUFilterMode.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUSamplerDescriptor.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUTextureFormat.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::Buffer::mapAsync):
(wgpuBufferGetUsage):
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::loadAction):
(WebGPU::storeAction):
(WebGPU::CommandEncoder::beginRenderPass):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(wgpuComputePassEncoderDispatchWorkgroups):
(wgpuComputePassEncoderDispatchWorkgroupsIndirect):
(wgpuComputePassEncoderEnd):
(wgpuComputePassEncoderDispatch): Deleted.
(wgpuComputePassEncoderDispatchIndirect): Deleted.
(wgpuComputePassEncoderEndPass): Deleted.
* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::baseFeatures):
(WebGPU::apple3):
(WebGPU::apple4):
(WebGPU::apple5):
(WebGPU::apple6):
(WebGPU::apple7):
(WebGPU::mac2):
(WebGPU::mergeLimits):
(WebGPU::defaultLimits):
* Source/WebGPU/WebGPU/QuerySet.mm:
(wgpuQuerySetGetCount):
(wgpuQuerySetGetType):
* Source/WebGPU/WebGPU/Queue.h:
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::Queue::onSubmittedWorkDone):
(wgpuQueueOnSubmittedWorkDone):
(wgpuQueueOnSubmittedWorkDoneWithBlock):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(wgpuRenderPassEncoderEnd):
(wgpuRenderPassEncoderEndPass): Deleted.
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::stepFunction):
* Source/WebGPU/WebGPU/Sampler.h:
(WebGPU::Sampler::isFiltering const):
* Source/WebGPU/WebGPU/Sampler.mm:
(WebGPU::validateCreateSampler):
(WebGPU::mipFilter):
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::findShaderModuleParameters):
(WebGPU::earlyCompileShaderModule):
(WebGPU::Device::createShaderModule):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::featureRequirementForFormat):
(WebGPU::isCompressedFormat):
(WebGPU::depthSpecificFormat):
(WebGPU::stencilSpecificFormat):
(WebGPU::Texture::texelBlockWidth):
(WebGPU::Texture::texelBlockHeight):
(WebGPU::isRenderableFormat):
(WebGPU::supportsMultisampling):
(WebGPU::hasStorageBindingCapability):
(WebGPU::Texture::removeSRGBSuffix):
(WebGPU::Texture::pixelFormat):
(WebGPU::Texture::texelBlockSize):
(WebGPU::Texture::depthOnlyAspectMetalFormat):
(WebGPU::Texture::stencilOnlyAspectMetalFormat):
(WebGPU::Texture::isValidImageCopySource):
(WebGPU::Texture::isValidImageCopyDestination):
(wgpuTextureGetDepthOrArrayLayers):
(wgpuTextureGetDimension):
(wgpuTextureGetFormat):
(wgpuTextureGetHeight):
(wgpuTextureGetWidth):
(wgpuTextureGetMipLevelCount):
(wgpuTextureGetSampleCount):
(wgpuTextureGetUsage):
* Source/WebGPU/WebGPU/WebGPU.h:
* Source/WebGPU/WebGPU/WebGPUExt.h:
* Source/WebKit/Shared/PALArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUFeatureName.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUSamplerDescriptor.h:
* Source/WebKit/Shared/WebGPU/WebGPUSamplerDescriptor.serialization.in:

Canonical link: <a href="https://commits.webkit.org/259608@main">https://commits.webkit.org/259608@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9da8d4c517c2f25580b0a686df878d4e61956a2a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105419 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14487 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38291 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114674 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174836 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109320 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15680 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5426 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97724 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114550 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111177 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95100 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93985 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26735 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81287 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7798 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28091 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7920 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4683 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13940 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47639 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6625 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9712 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

litherum commented on Source/WebCore/Modules/WebGPU/GPUAdapter.cpp:
```diff
@@ -65,4 +65,9 @@ void GPUAdapter::requestDevice(ScriptExecutionContext&, const std::optional<GPUD
 
+void GPUAdapter::requestAdapterInfo(ScriptExecutionContext&, const std::optional<Vector<String>>&, RequestAdapterInfoPromise&& promise)
+{
+    promise.resolve(nullptr);
```
`// FIXME`?

mwyrzykowski replied:
Good idea

litherum commented on Source/WebCore/Modules/WebGPU/GPUAdapter.idl:
```diff
@@ -37,4 +37,5 @@ interface GPUAdapter {
     readonly attribute boolean isFallbackAdapter;
 
     [CallWith=CurrentScriptExecutionContext] Promise<GPUDevice> requestDevice(optional GPUDeviceDescriptor descriptor);
+    [CallWith=CurrentScriptExecutionContext] Promise<undefined> requestAdapterInfo(optional sequence<DOMString> unmaskHints = []);
```
Do we really need the script execution context for this?

mwyrzykowski replied:
Not at this point, I will remove it

litherum commented on Source/WebGPU/WebGPU/WebGPUExt.h:
```diff
@@ -207,7 +190,7 @@ WGPU_EXPORT bool wgpuDevicePopErrorScopeWithBlock(WGPUDevice device, WGPUErrorBl
 WGPU_EXPORT void wgpuDeviceSetUncapturedErrorCallbackWithBlock(WGPUDevice device, WGPUErrorBlockCallback callback);
 WGPU_EXPORT void wgpuInstanceRequestAdapterWithBlock(WGPUInstance instance, WGPURequestAdapterOptions const * options, WGPURequestAdapterBlockCallback callback);
-WGPU_EXPORT void wgpuQueueOnSubmittedWorkDoneWithBlock(WGPUQueue queue, uint64_t signalValue, WGPUQueueWorkDoneBlockCallback callback);
+WGPU_EXPORT void wgpuQueueOnSubmittedWorkDoneWithBlock(WGPUQueue queue, WGPUQueueWorkDoneBlockCallback callback);
```
YEEESSSSSSS

litherum commented on Source/WebGPU/WebGPU/WebGPU.h:
```diff
@@ -216,29 +216,32 @@ typedef enum WGPUDeviceLostReason {
 typedef enum WGPUErrorFilter {
```
I assume this file is literally unchanged from upstream?

mwyrzykowski replied:
Yes that is correct

litherum commented on Source/WebGPU/WebGPU/WebGPU.h:
```diff
@@ -839,30 +870,33 @@ typedef struct WGPUSamplerBindingLayout {
 } WGPUSamplerDescriptor;
 
-typedef struct WGPUShaderModuleDescriptor {
+typedef struct WGPUShaderModuleCompilationHint {
```
Sweeeeet

litherum commented on Source/WebGPU/WebGPU/Buffer.mm:
```diff
@@ -389,3 +389,9 @@ void wgpuBufferSetLabel(WGPUBuffer buffer, const char* label)
+
+WGPUBufferUsage wgpuBufferGetUsage(WGPUBuffer buffer)
+{
+    // FIXME: this shouldn't need a cast - https://github.com/webgpu-native/webgpu-headers/issues/172
+    return static_cast<WGPUBufferUsage>(WebGPU::fromAPI(buffer).usage());
```
should we make fromAPI() with a bunch of overloads for these?

mwyrzykowski replied:
Maybe I misunderstand the suggestion but I don't know if we can, since the argument would be a uint32_t (WGPUBufferUsageFlags is typedef'ed as uint32_t)

litherum replied:
awww bummer. ok

litherum commented on Source/WebGPU/WebGPU/CommandEncoder.mm:
```diff
@@ -46,9 +46,10 @@ static MTLLoadAction loadAction(WGPULoadOp loadOp)
+    case WGPULoadOp_Undefined:
     case WGPULoadOp_Force32:
         ASSERT_NOT_REACHED();
-        return MTLLoadActionClear;
+        return MTLLoadActionDontCare;
```
I'm kind of confused about this; it doesn't match the WebGPU spec IDL. Maybe I should open an issue on WebGPU.h about it....

mwyrzykowski replied:
Yes I will do that, unclear why `WGPULoadOp_Undefined` was added

mwyrzykowski replied:
@litherum as I was typing up the WebGPU issue I noticed the `_Undefined` categories exist for many (all?) of the other enums, it seems like it was an oversight not having it before

litherum commented on Source/WebGPU/WebGPU/QuerySet.mm:
```diff
@@ -136,3 +136,13 @@ void wgpuQuerySetSetLabel(WGPUQuerySet querySet, const char* label)
+
+WGPUQueryType wgpuQuerySetGetType(WGPUQuerySet querySet)
+{
+    return WebGPU::fromAPI(querySet).visibilityBuffer() ? WGPUQueryType_Occlusion : WGPUQueryType_Timestamp;
```
I think we should make a QuerySet::getType() instead of doing logic here. I don't think the `wgpu` functions themselves should do anything at all except API marshaling.

mwyrzykowski replied:
Good idea, I agree

litherum commented on Source/WebGPU/WebGPU/Texture.mm:
```diff
@@ -2900,3 +2872,44 @@ void wgpuTextureSetLabel(WGPUTexture texture, const char* label)
+
+uint32_t wgpuTextureGetDepthOrArrayLayers(WGPUTexture texture)
+{
+    return WebGPU::fromAPI(texture).descriptor().size.depthOrArrayLayers;
```
Ditto about making `Texture.depthOrArrayLayers()`, and similar elsewhere too

litherum commented on Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUQueueImpl.h:
```diff
@@ -84,8 +84,6 @@ class QueueImpl final : public Queue {
 
     void setLabelInternal(const String&) final;
 
-    uint64_t m_signalValue { 1 };
```
Eeeeeexcellent :D

litherum reviewed (approved):
Thank you!!!

mwyrzykowski committed:
[WebGPU] Update WebGPU.h to latest revision - fa1c6ab4927ef1fa5731907b42b62ea93119866c (250856)
https://bugs.webkit.org/show_bug.cgi?id=250856
<radar://104437950>

Reviewed by Myles C. Maxfield.

Pull the latest version of WebGPU.h and update our
sources to use it.

Significant changes:
 * Shader hints moved into WebGPU.h, previously in WebGPUExt.h
 * MipmapFilter became its own type
 * GPUAdapter has requestAdapterInfo

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/WebGPU/GPUAdapter.cpp:
(WebCore::GPUAdapter::requestAdapterInfo):
* Source/WebCore/Modules/WebGPU/GPUAdapter.h:
* Source/WebCore/Modules/WebGPU/GPUAdapter.idl:
* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUFeatureName.h:
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUFeatureName.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerDescriptor.h:
* Source/WebCore/Modules/WebGPU/GPUSamplerDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureFormat.h:
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUTextureFormat.idl:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUAdapterImpl.cpp:
(PAL::WebGPU::supportedFeatures):
(PAL::WebGPU::AdapterImpl::requestDevice):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUComputePassEncoderImpl.cpp:
(PAL::WebGPU::ComputePassEncoderImpl::dispatch):
(PAL::WebGPU::ComputePassEncoderImpl::dispatchIndirect):
(PAL::WebGPU::ComputePassEncoderImpl::end):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.cpp:
(PAL::WebGPU::ConvertToBackingContext::convertToBacking):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.cpp:
(PAL::WebGPU::createBackingDescriptor):
(PAL::WebGPU::DeviceImpl::createShaderModule):
(PAL::WebGPU::DeviceImpl::popErrorScope):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUQueueImpl.cpp:
(PAL::WebGPU::QueueImpl::onSubmittedWorkDone):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUQueueImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPURenderPassEncoderImpl.cpp:
(PAL::WebGPU::RenderPassEncoderImpl::end):
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUFeatureName.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUFilterMode.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUSamplerDescriptor.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUTextureFormat.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::Buffer::mapAsync):
(wgpuBufferGetUsage):
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::loadAction):
(WebGPU::storeAction):
(WebGPU::CommandEncoder::beginRenderPass):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(wgpuComputePassEncoderDispatchWorkgroups):
(wgpuComputePassEncoderDispatchWorkgroupsIndirect):
(wgpuComputePassEncoderEnd):
(wgpuComputePassEncoderDispatch): Deleted.
(wgpuComputePassEncoderDispatchIndirect): Deleted.
(wgpuComputePassEncoderEndPass): Deleted.
* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::baseFeatures):
(WebGPU::apple3):
(WebGPU::apple4):
(WebGPU::apple5):
(WebGPU::apple6):
(WebGPU::apple7):
(WebGPU::mac2):
(WebGPU::mergeLimits):
(WebGPU::defaultLimits):
* Source/WebGPU/WebGPU/QuerySet.mm:
(wgpuQuerySetGetCount):
(wgpuQuerySetGetType):
* Source/WebGPU/WebGPU/Queue.h:
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::Queue::onSubmittedWorkDone):
(wgpuQueueOnSubmittedWorkDone):
(wgpuQueueOnSubmittedWorkDoneWithBlock):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(wgpuRenderPassEncoderEnd):
(wgpuRenderPassEncoderEndPass): Deleted.
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::stepFunction):
* Source/WebGPU/WebGPU/Sampler.h:
(WebGPU::Sampler::isFiltering const):
* Source/WebGPU/WebGPU/Sampler.mm:
(WebGPU::validateCreateSampler):
(WebGPU::mipFilter):
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::findShaderModuleParameters):
(WebGPU::earlyCompileShaderModule):
(WebGPU::Device::createShaderModule):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::featureRequirementForFormat):
(WebGPU::isCompressedFormat):
(WebGPU::depthSpecificFormat):
(WebGPU::stencilSpecificFormat):
(WebGPU::Texture::texelBlockWidth):
(WebGPU::Texture::texelBlockHeight):
(WebGPU::isRenderableFormat):
(WebGPU::supportsMultisampling):
(WebGPU::hasStorageBindingCapability):
(WebGPU::Texture::removeSRGBSuffix):
(WebGPU::Texture::pixelFormat):
(WebGPU::Texture::texelBlockSize):
(WebGPU::Texture::depthOnlyAspectMetalFormat):
(WebGPU::Texture::stencilOnlyAspectMetalFormat):
(WebGPU::Texture::isValidImageCopySource):
(WebGPU::Texture::isValidImageCopyDestination):
(wgpuTextureGetDepthOrArrayLayers):
(wgpuTextureGetDimension):
(wgpuTextureGetFormat):
(wgpuTextureGetHeight):
(wgpuTextureGetWidth):
(wgpuTextureGetMipLevelCount):
(wgpuTextureGetSampleCount):
(wgpuTextureGetUsage):
* Source/WebGPU/WebGPU/WebGPU.h:
* Source/WebGPU/WebGPU/WebGPUExt.h:
* Source/WebKit/Shared/PALArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUFeatureName.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUSamplerDescriptor.h:
* Source/WebKit/Shared/WebGPU/WebGPUSamplerDescriptor.serialization.in:

Canonical link: https://commits.webkit.org/259608@main
